File: 1768-merge-strings-alternately/1768-merge-strings-alternately.js
/**
 * @param {string} word1
 * @param {string} word2
 * @return {string}
 */
var mergeAlternately = function(word1, word2) {
    let ansStr = [];
    
    const greater = Math.max(word1.length, word2.length);
    
    for(let i = 0; i < greater; i++){
        ansStr.push(word1[i]);
        ansStr.push(word2[i]);
    }
    
    return ansStr.join("");
};

File: 1768-merge-strings-alternately/1768-merge-strings-alternately.go
func mergeAlternately(word1 string, word2 string) string {
    ansStr := []string{}
    
    greater := 0
    
    if len([]rune(word1)) > len([]rune(word2)){
        greater = len([]rune(word1))
    }else{
        greater = len([]rune(word2))
    }
    
    for i := 0; i < greater; i++{
        if i < len([]rune(word1)){
            ansStr = append(ansStr, string([]rune(word1)[i]))
        }
        
        if i < len([]rune(word2)){
            ansStr = append(ansStr, string([]rune(word2)[i]))
        }
    }
    
    return strings.Join(ansStr, "")
}